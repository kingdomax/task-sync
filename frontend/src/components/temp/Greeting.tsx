interface GreetingProps {
    name: string;
}

export const Greeting = ({ name }: GreetingProps) => {
    return <h1>Hello {name}</h1>;
};
