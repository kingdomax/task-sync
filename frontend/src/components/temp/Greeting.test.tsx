import { render } from '@testing-library/react';

import { Greeting } from './Greeting';

describe('<Greeting />', () => {
    test('greeting renders and match snapshot', () => {
        const { container, getByText } = render(<Greeting name="Pramoch" />);
        expect(container).toMatchSnapshot(); // structure test
        expect(getByText('Hello Pramoch')).toBeInTheDocument(); // behavior test
    });
});
