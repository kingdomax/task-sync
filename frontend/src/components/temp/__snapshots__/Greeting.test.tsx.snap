// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dummy test 1`] = `
<div>
  <h1>
    Hello 
    Pramoch
  </h1>
</div>
`;

exports[`greeting renders and match snapshot 1`] = `
<div>
  <h1>
    Hello 
    Pramoch
  </h1>
</div>
`;
